pkg/doc/verifiable/common.go

```diff
@@ -83,27 +83,12 @@ func (r *DIDKeyResolver) resolvePublicKey(issuerDID, keyID string) (*verifier.Pu
 		return nil, fmt.Errorf("resolve DID %s: %w", issuerDID, err)
 	}
 
-	for _, key := range doc.PublicKey {
-		// TODO remove string contains after sidetree create public key with this format DID#KEYID
-		// sidetree now return #KEYID
-		if strings.Contains(key.ID, keyID) {
+	for _, vm := range doc.VerificationMethods() {
```

Comment:
Need a quick followup to only allow certain VerificationMethods.

Reply:
also what if the key isnt in VerificationMethods?

Reply:
Added:
1) `Relationship` parameter to `VerificationMethod`.
2) `customVerificationRelationships` variadic parameter to `DID.VerificationMethods()`.
3) Special `VerificationRelationshipGeneral` for the public keys that are not referred from any Verification Method.